lib/helpers.js

```diff
@@ -30,6 +30,34 @@ var formatValidators = require('./validation/format-validators');
 var mocker = require('json-schema-faker');
 var ZSchema = require('z-schema');
 
+// full-date from http://xml2rfc.ietf.org/public/rfc/html/rfc3339.html#anchor14
+var dateRegExp = new RegExp(
+  '^' +
+  '\\d{4}' + // year
+  '-' +
+  '([0]\\d|1[012])' + // month
+  '-' +
+  '(0[1-9]|[12]\\d|3[01])' + // day
+  '$');
+  
+// date-time from http://xml2rfc.ietf.org/public/rfc/html/rfc3339.html#anchor14
+var dateTimeRegExp = new RegExp(
```

Comment:
I hope these regex breakdowns make sense and are helpful. It's better than having 128+ characters spewed out on one line IMO.